numbers/distribution.go

```diff
@@ -224,75 +228,76 @@ func PoissonSum(left int, right int, lambda float64) float64 {
 }
 
 // BinomialLeftSummation calculates the sum of binomial probabilities to the left of k successes for a binomial distribution with n experiments and a success probability of p, inclusive.
-func BinomialLeftSummation(n int, k int, p float64) float64 {
-	if k <= n/2 {
-		return evaluateLeftBinomialSum(n, k, p)
-	} else {
-		return 1 - evaluateRightBinomialSum(n, k+1, p)
+func BinomialLeftSummation(n int, k int, p float64, logOutput bool) float64 {
+	if n == k {
+		if logOutput {
+			return 0
+		} else {
+			return 1
+		}
 	}
+	return evaluateLeftBinomialSum(n, k, p, logOutput)
 }
 
 // BinomialRightSummation calculates the sum of binomial probabilities to the right of k successes for a binomial distribution with n experiments and a success probability of p, inclusive.
-func BinomialRightSummation(n int, k int, p float64) float64 {
-	if k > n/2 {
-		return evaluateRightBinomialSum(n, k, p)
-	} else {
-		return 1 - evaluateLeftBinomialSum(n, k-1, p)
+func BinomialRightSummation(n int, k int, p float64, logOutput bool) float64 {
+	if n == k {
+		return BinomialDistLog(n, n, p)
+	}
+	if k == 0 {
+		if logOutput {
+			return 0
+		} else {
+			return 1
+		}
 	}
+	return evaluateRightBinomialSum(n, k, p, logOutput)
 }
 
 // BinomialSum calculates the sum of probabilities in a binomial distribution with n experiments and success probability p between two input k values. Inclusive on both ends.
-func BinomialSum(left int, right int, n int, p float64) float64 {
+func BinomialSum(left int, right int, n int, p float64, logOutput bool) float64 {
 	if right < left {
 		log.Fatalf("BinomialSum failed. Right side value must be greater than the left side value.")
 	}
-	var answer float64 = 0
+	var answer, _ = BinomialDist(n, left, p, logOutput)
 	var curr float64
 	for i := left; i <= right; i++ {
-		curr, _ = BinomialDist(n, i, p)
-		answer += curr
+		curr, _ = BinomialDist(n, i, p, logOutput)
+		if logOutput {
+			answer = logspace.Add(answer, curr)
+		} else {
+			answer += curr
```

Comment:
Is there an advantage to doing all this math in non-log-space?  Seems like we could do it all in log-space and then just convert at the end?  Maybe that is slower since logspace.Add is slower than normal addition?

Reply:
This comment also applies to some of the functions below, but the more I think about it the less clear I am that we want to be in log-space if we will eventually get out of it and have to do lots of addition operations

Reply:
Yes, exactly like you said, logspace.Add is significantly slower than addition. Normal space calculations are therefore much faster. The big question is whether the user is concerned if an area under the curve is reported as 0 if we hit underflow. If they don't care, the regular space calculation will be much faster. Note that when logOutput is true, we never get out of logSpace, so we're not doing all the logspace.Add operations for nothing